framework/graph/Graph.kt

```diff
@@ -46,7 +46,7 @@ class Graph(private val interactions: Interactions) {
 
     private val _thingVertices: MutableMap<String, Vertex.Thing> = ConcurrentHashMap()
     private val _typeVertices: MutableMap<String, Vertex.Type> = ConcurrentHashMap()
-    private val _edges: MutableList<Edge> = Collections.synchronizedList(mutableListOf())
+    private val _edges: MutableSet<Edge> = Collections.synchronizedSet(mutableSetOf())
```

Comment:
We never want to list the exact same edge (with the same source vertex and target vertex) more than once, so the best representation of this collection is a set.